docs: user tutorial for GCP NFS

abalaie opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow contributing guidelines, templates, the recommended Git workflow, and any related documentation.
2. Read and submit the required Contributor Licence Agreements (https://github.com/kyma-project/community/blob/main/CONTRIBUTING.md#agreements-and-licenses).
3. Test your changes and attach their results to the pull request.
4. Update the relevant documentation.

If the pull request requires a decision, follow the [decision-making process](https://github.com/kyma-project/community/blob/main/governance.md) and replace the PR template with the [decision record template](https://github.com/kyma-project/community/blob/main/.github/ISSUE_TEMPLATE/decision-record.md).
-->

**Description**

Changes proposed in this pull request:

- user tutorial for GCP NFS
- small changes to glossary
- small changes to gcpNfsVolume CRD document

**Related issue(s)**
<!-- If you refer to a particular issue, provide its number. For example, `Resolves #123`, `Fixes #43`, or `See also #33`. -->

abalaie commented:
Updated the document to be in sync with recent changes.

mmitoraj commented on docs/user/tutorials/01-20-gcp-nfs-volume.md:
````diff
@@ -0,0 +1,190 @@
+   kubectl logs -n $NAMESPACE `kubectl get pod -n $NAMESPACE -l app=gcpnfsvolume-demo -o=jsonpath='{.items[0].metadata.name}'`
+   ```
+   
+   The command should print something like:
````
```suggestion
   The command should print a result similar to the following one:
```

mmitoraj commented on docs/user/tutorials/01-20-gcp-nfs-volume.md:
````diff
@@ -0,0 +1,190 @@
+   ...
+   ```
+   
+   Note that the content after `File content:` contains prints from both workloads. This 
````
```suggestion
   >[!NOTE] The `File content:` contains prints from both workloads. This demonstrates the ReadWriteMany capability of the volume.
```

mmitoraj commented on docs/user/tutorials/01-20-gcp-nfs-volume.md:
````diff
@@ -0,0 +1,190 @@
+   ```
+   
+   Note that the content after `File content:` contains prints from both workloads. This 
+   demonstrates the ReadWriteMany capability of the volume.
````
```suggestion
```

mmitoraj commented on docs/user/tutorials/01-20-gcp-nfs-volume.md:
```diff
@@ -0,0 +1,190 @@
+
+   Similarly to PV, note the `RWX` access mode and `Bound` status.
+
+6. Create two workloads that both write to the volume
```
```suggestion
6. Create two workloads that both write to the volume. Run:
```

mmitoraj reviewed (changes requested):
Please also add the tutorial to the list in `_sidebar.md`.

abalaie commented:
I rebased the code as it had some conflicts with the base and the option was not available on the UI. 
But then it showed many commits that were already merged to the "main" branch. So based on Github suggestions on the same issue, I changed the base branch to something else and again back to main, so that the base branch commit get updated and it only shows the relevant changes. And it worked.
